client/tx/tx.go

```diff
@@ -94,16 +94,19 @@ func BroadcastTx(clientCtx client.Context, txf Factory, msgs ...sdk.Msg) error {
 		}
 
 		if err := clientCtx.PrintRaw(json.RawMessage(txBytes)); err != nil {
-			_, _ = fmt.Fprintf(os.Stderr, "%s\n", txBytes)
+			_, _ = fmt.Fprintf(os.Stderr, "error: %v\n%s\n", err, txBytes)
```

Comment:
Do we really want `%s` formatter for `[]byte` that reflect an encoded tx? I would rather print the hash or JSON-ified tx.

Reply:
@alexanderbez thank you for the post! Perhaps not but that's out of the scope of this PR and for that I've filed https://github.com/cosmos/cosmos-sdk/issues/14314

Reply:
```bash
$ simd tx bank send $BOB $ALICE 100stake
error: yaml: found unexpected end of stream
{"body":{"messages":[{"@type":"/cosmos.bank.v1beta1.MsgSend","from_address":"cosmos1xxgl377yujn4pkmvp3yk85tz3kyzd58zzxr8f8","to_address":"cosmos1swwj7v8d4cw6rhs8mu829fvcvhgdr5qvcvyc3c","amount":[{"denom":"stake","amount":"100"}]}],"memo":"","timeout_height":"0","extension_options":[],"non_critical_extension_options":[]},"auth_info":{"signer_infos":[],"fee":{"amount":[],"gas_limit":"200000","payer":"","granter":""},"tip":null},"signatures":[]}
confirm transaction before signing and broadcasting [y/N]: 
```

Actually, it prints the jsonified tx in case of error. I've tweaked it a bit so it always returns an error and look above.